decentraland/unity-renderer
# feat: [BASE BRANCH] Lazy loading for social data

## What does this PR change?
Load all the social data like private messages, friend list and friend requests in a lazy way.

An example: user opens the friends window from the taskbar, only then, the friend list is requested from the renderer to kernel and then to the servers.

### Implementation parts
- [x] Friends initialization
- [x] Friend list is displayed when the user opens the window
- [x] Friend request list is displayed when the users opens the tab
- [x] Conversation list is displayed
- [x] A private chat window with a specific friend is opened

## Documentation
**RFC**: https://www.notion.so/decentraland/RFC-Lazy-loading-for-social-data-06ff22ce0545464b83da3e4b0457e901

## How to test the changes?
1. Go to: https://play.decentraland.org/?kernel-branch=feat/friends-lazy-load-only&renderer-branch=feat/social-lazy-loading
2. Write & receive private messages
3. Write & receive messages in #nearby
4. Check the conversation list
5. Check the friend list
6. Check that the notifications in the taskbar are displayed correctly whenever you receive a friend request or new messages are received
7. Send & receive friend requests
8. Load the chat history of a private conversation
9. Search for friends in the conversation list

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
